orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -114,6 +115,12 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
         transformers.emplace_back(onnxruntime::make_unique<TransformerLayerRecompute>(
             config.number_recompute_layers, compatible_eps));
       }
+      if (config.propagate_cast_ops_level >= 0) {
+        std::unordered_set<std::string> cuda_execution_provider = {onnxruntime::kCudaExecutionProvider};
```

Comment:
This transformer is applicable to ROCm EP as well.

Reply:
@SherlockNoMad though we should limit this version only to CUDA ROCm needs more consideration.

Reply:
@weixingzhang, let's add Rocm support in latter iterations.